timeskew/timeskew.go

```diff
@@ -0,0 +1,117 @@
+package timeskew
+
+import (
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/beevik/ntp"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	// DefaultServer will be internally resolved to the closest available.
+	// also it rarely queries same server more than once.
+	DefaultServer = "pool.ntp.org"
+
+	// DefaultAttempts defines how many servers we will query
+	DefaultAttempts = 3
+
+	// DefaultUpdatePeriod defines how often time will be queried from ntp.
+	DefaultUpdatePeriod = 2 * time.Minute
+)
+
+type ntpQuery func(string) (*ntp.Response, error)
+
+func computeOffset(queryMethod ntpQuery, server string, attempts int) (time.Duration, error) {
+	offsets := make([]time.Duration, 5)
+	// lowest and highest will be cutoff to prevent reading data from out of sync ntp servers
+	for i := 0; i < attempts+2; i++ {
```

Comment:
With channels and goroutines, you can easily send all requests at once.

Reply:
thanks, slipped from my mind